### DISCUSSION TITLE
Overwhelmed.

### DISCUSSION THREAD

**Discussion by luckman212** [brianpetro/obsidian-smart-connections#692] [Q&A]
After seeing Smart Connections (SC) hit over 200K installs (congrats! 🎉 ) over this long weekend, I finally decided to give it a go.

I'm just a dabbler when it comes to AI, LLMs, and all the other jargon that comes with them: Embeddings, Tokens, OpenRouter, etc. So my first reaction when opening up the settings was... _"Holy $#!&"_

After randomly choosing what seemed like the best options (free? sure I'll take it... 😐) I let SC loose on my vault as it started churning though my 3,000+ note vault. _**I panicked initially**_, not really sure if it was uploading all of my notes to OpenAI or some other cloud model (I don't think it does, at least not by default?)

I tried looking around, read the [README][2] several times, watched a [YouTube vid][1], and then admitted defeat. I'm left with questions...

1. Is there any guidance for what models to choose for Notes/Blocks/Chat? I have an OpenAI API key + ChatGPT Plus subscription. I don't generally use Claude, Meta or local LLMs—but I would like to if they yield better results for SC, or open up pathways to more privacy. I've chosen `BGE-micro-v2` for Notes/Blocks and `meta-llama/llama-3-8b-instruct:free` for Chat. But for no particular reason!

2. Is it OK to sync the hidden `.smart-env` folder in my vault? I use Obsidian on 3 computers + my mobile phone. Is there anything dangerous about syncing this folder? I'm not sure what would happen if SC tries to write data / process the same files from 2 different computers. I can imagine a mess.

3. I'm closely following #670 because having my Chats pile up in my vault with what amounts to throwaway info for me is triggering my OCD.

Thanks for any tips for a beginner 🙏 

[1]: https://www.youtube.com/watch?v=cgW7BpXURUQ
[2]: https://github.com/brianpetro/obsidian-smart-connections?tab=readme-ov-file

**Comment by hittjwX** [comment]
Start with the smallest possible local model and most basic configuration. The `BGE-tiny` and `BGE-small-4K` work well enough to get familiar with the tool. Use a 50 Minimum Embedding Lengths and 248 Local Embedding Max Tokens. 

The *Open Router* `openchat/openchat-7b:free` helps you process. You'll need an API key. Instructions are available in the Smart Connections settings. 

I don't sync the `.smart-connections` folder and use a different profile for clients that don't have a GPU when it comes to Obsidian. I have a writing, research, mobile, and editing computer.

The editing computer has more horse power. There is nothing dangerous in these folders, but I don't want to spend the time on a laptop in a remote office rebuilding the LLM details.

As far as I can tell, the only outside connection is to Open Router when using a free local model. I've looked at packets to confirm this. However, it's hard to tell if you were using an Ollama server or different models.

I get concerned sometimes when the LLM knows stuff that is not in my notes, or infers something with a high level of accuracy that would require outside information. My focus is on working faster than someone who might access my notes.

I'm a publisher of newsletters, podcast, and other publications that touch on social engineering, anti-money laundering, strategic decision making, and business development. If anyone knows how else I can check to make sure data doesn't leave my system -- or train another model, please let me know.

If you are using this for business and have a monetary upside, I've been helping independent consultants, subject matter experts, and thought leaders become more efficient with their note-taking. I'll be happy to answer any questions here.
